src/utility/test_tools.hpp

```diff
@@ -84,6 +84,118 @@ template <class mtype> class SubscriberTB
     std::vector<mtype> m_msgs;
     ros::Subscriber m_sub;
 };
+
+//this class is similar to the SubscriberTB, but is meant for analyzing a
+//1-D stream of data. The format_callback should be a function that takes
+//in the message datatype, and returns 1 double value, which is stored and
+//later analyzed.
+template <class mtype> class SubscriberAnalyzer
+{
+public:
+    SubscriberAnalyzer()
+    {}
+    ~SubscriberAnalyzer()
+    {}
+    void Init(std::string topic,
+              double (*format_callback)(const typename mtype::ConstPtr&))
+    {
+        ros::NodeHandle n;
+
+        m_sub = n.subscribe(topic.c_str(), 1,
+                            &SubscriberAnalyzer<mtype>::Callback, this);
+
+        m_format_function = format_callback;
+
+        m_enabled = false;
+    }
+
+    void Callback(const typename mtype::ConstPtr& msg)
+    {
+        if(m_enabled == true)
+        {
+            double data = m_format_function(msg);
+            m_data.push_back(data);
+        }
+    }
+
+    void Start()
+    {
+        m_enabled = true;
+    }
+
+    void Stop()
+    {
+        m_enabled = false;
+    }
+
+    void ClearData()
+    {
+        m_data.clear();
+    }
+
+    double GetMin()
+    {
+        if(m_data.size() == 0)
+        {
+            ROS_ERROR("no data collected, can't calculate min!");
+            return 0.0;
+        }
+
+        double min = m_data[0];
+        for (int i = 0; i < m_data.size(); ++i)
```

Comment:
Since you've already checked the zeroth element, `i` can start at 1. Minor detail.